clinlp/entity.py

```diff
@@ -213,3 +214,41 @@ def __call__(self, doc: Doc):
         doc.set_ents(entities=ents)
 
         return doc
+
+
+def create_term(row, col_term):
+    """Creates a Term object with available attributes."""
+    term = Term("")
+    for attr, value in row.items():
+        if attr == col_term:
+            term.phrase = value
+        if value == value:
+            if value == "True":
+                value = True
+            elif value == "False":
+                value = False
+            elif isinstance(value, float):
+                value = int(value)
+            setattr(term, attr, value)
+    return term
+
+def concept_dict_creator(data, col_label, col_term):
+    """Transforms source concept data to a dictionary that the clinlp entity matcher can read.
+    Takes a csv file where each row is a distinct word or sentence (term) that belongs to an entity label (label)."""
+
+    df = pd.read_csv(data)
+    # Makes Terms of rows where an attribute is specified
+    ## Make a list of all attribute columns, assuming the first two columns are entity label and term
+    attr_cols = list(df.columns[2:])
+    ## Find rows where one or more attribute columns are filled and replace the value in the 'term' column with a Term class with these attributes
+    df.loc[df[attr_cols].notna().any(axis=1), col_term] = df[df[attr_cols].notna().any(axis=1)].apply(create_term, col_term=col_term, axis=1)
+    ## Drop the attribute columns
+    df.drop(columns=attr_cols, inplace=True)
```

Comment:
Waarom moeten deze verwijderd worden? Voor de groupby? Dan zou ik ze erin laten, en in de groupby specificeren welke kolommen gegroepeerd moeten worden.